spec/controllers/rents_controller_spec.rb

```diff
@@ -0,0 +1,175 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe RentsController do
+  include_context 'Authenticated User'
+
+  describe 'GET #create' do
+    context 'When creating a rent for an user' do
+      let(:user) { create(:user) }
+      let(:book) { create(:book) }
+      let(:rent) { attributes_for(:rent, book_id: book.id) }
+      before do
+        post :create, params: { user_id: user.id, rent: rent }
+      end
+
+      it 'returns http status created' do
+        expect(response).to have_http_status(:created)
+      end
+    end
+
+    context 'When given a non existent user ID' do
+      let(:book) { create(:book) }
+      let(:rent) { attributes_for(:rent, book_id: book.id) }
+      before do
+        post :create, params: { user_id: User.last.id + 1, rent: rent }
+      end
+
+      it 'returns http status 404' do
+        expect(response).to have_http_status(:not_found)
+      end
+
+      it 'responds with an error message' do
+        expect(response_body).to have_key('error')
+      end
+    end
+
+    context 'When there is a missing param' do
```

Comment:
this test is testing book_id missing param, change the context or add sub contexts with tests for each attribute